src/ipc.rs

```diff
@@ -735,14 +724,17 @@ impl OpaqueIpcMessage {
                         &mut *os_ipc_channels_for_deserialization.borrow_mut(),
                         &mut self.os_ipc_channels,
                     );
-                    mem::swap(
+                    let old_ipc_shared_memory_regions_for_deserialization = mem::replace(
                         &mut *os_ipc_shared_memory_regions_for_deserialization.borrow_mut(),
-                        &mut self.os_ipc_shared_memory_regions,
+                        self.os_ipc_shared_memory_regions
+                            .into_iter()
+                            .map(Some)
+                            .collect(),
                     );
                     let result = bincode::deserialize(&self.data[..]);
-                    mem::swap(
+                    let _ = mem::replace(
                         &mut *os_ipc_shared_memory_regions_for_deserialization.borrow_mut(),
-                        &mut self.os_ipc_shared_memory_regions,
+                        old_ipc_shared_memory_regions_for_deserialization,
```

Comment:
I think this would be equivalent to (and more clear as):
```rust
*os_ipc_shared_memory_regions_for_deserialization.borrow_mut() = old_ipc_shared_memory_regions_for_deserialization;
```

Reply:
:+1: